src/App.java

```diff
@@ -1,5 +1,203 @@
+                 * breaks out of the loop, moving onto the next part
+                 */
+                System.out.println(myRandom.Randomization(studentsPerTable));
+                System.out.println("Would you like to re-randomize?");
+                option2 = input.nextLine();
+            }
+            System.out.println("Would you like to save this to a file?");
+            option2 = input.nextLine();
+            // this is needed because all of the other inputs have been an int, so this
+            // clarifies that the next one will be a String, not an int
+            while (option2.indexOf("ye") > -1 && fileError == true || option2.indexOf("Ye") > -1 && fileError == true) {
+                System.out.println("What would you like to name this file?");
+                filename = input.nextLine() + ".txt";
+                // this sends the filename chosen by the user into the method, which saves the
+                // groups into a new file
+                fileError = myRandom.saveToFile(filename);
+                // fileError is updated to be false (if there's no error) or if there is an
+                // error, it remains true, not breaking out of the loop
+            }
+            openingMenu(); 
+
+        } else if (option == 2) {
+            System.out.println("Which class' names would you like to edit?");
+            input.nextLine();
+            filename = input.nextLine() + ".txt"; 
+            myRandom = readBlock(filename);
+            System.out.println("Would you like to remove or add?");
+            option2 = input.nextLine();
+            if (option2.indexOf("remove") > -1 && option2.indexOf("add") == -1 || option2.indexOf("Remove") > -1 && option2.indexOf("Add") == -1){ //add and Add count as different answers so I need to account for both of them
+                System.out.println("What names would you like to remove from this file?");
+                names = input.nextLine();
+                myRandom.deleteNames(names + "//");
+                myRandom.editFile(filename);
+            }else if (option2.indexOf("remove") == -1 && option2.indexOf("add") > -1 || option2.indexOf("Remove") == -1 && option2.indexOf("Add") > -1){
+                System.out.println("What names would you like to add? Please seperate them by a // and no spaces");
+                names = input.nextLine();
+                myRandom.editArray(names + "//", filename); //I need to add "//" at the end of the string of names because otherwise the editArray method will stop at the second to last name, since it doesn't see a // at the end of the string
+               
+            }else if (option2.indexOf("remove") == -1 && option2.indexOf("add") == -1 || option2.indexOf("Remove") == -1 && option2.indexOf("Add") == -1) {
+                System.out.println("Please say whether you would like to remove or add names. Returning to opening menu.");
+                openingMenu();

```

Comment:
calling a method from itself is called a *recursive procedure* and is powerful, but not really ideal for situations like this - it means that if you run the menu five or six times technically all five or six copies of the `openingMenu` method are running. It would be better to either wrap the entire method body in a loop OR wrap the call of the method in a loop (in start) to manage the re-running of the method.